src/codeu/chat/server/fileWriter.java

```diff
@@ -0,0 +1,98 @@
+package codeu.chat.server;
+
+import java.util.concurrent.BlockingQueue;
+import java.lang.Runnable;
+
+import codeu.chat.common.Writeable;
+import codeu.chat.util.Serializers;
+
+import java.io.FileOutputStream;
+import java.io.FileInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import codeu.chat.common.User;
+
+import codeu.chat.util.Logger;
+
+public class fileWriter implements Runnable {
+
+    private static BlockingQueue<Writeable> queue;
+    public static final String TRANSACTION_FILE = "transaction.log";
+
+    public static final String CNT_FILE = "transaction.cnt";
```

Comment:
I'm not convinced you need this file. I assume the main use is so you know when to stop reading the transaction file? If that's the case, look more closely at the file input stream API. The stream can tell you when it's reached the end of the file.